### PR TITLE
feat: inputs: add readonly support and its styles

### PR DESCRIPTION
**Pull request by dkilgore-eightfold**
## SUMMARY:

- Updates `TextInput`, `TextArea`, `SearchBox` `readonly` and `readOnlyProps`, enabling complex custom readonly implementations.
- Updates Select `readonly` to behave as expected
- Adds UTs

https://github.com/EightfoldAI/octuple/assets/99700808/75995ec6-fa42-49e3-ba8f-82bcfa0c5f29

## JIRA TASK (Eightfold Employees Only):
ENG-60991

## CHANGE TYPE:

- [x] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [x] I have added unittests for this change

## TEST PLAN:
Pull the PR branch and run `yarn` and `yarn storybook`. Verify the `Inputs` and `Select` stories behave as expected.

### CODE REVIEW

**Review by ypike-eightfold — APPROVED** [review]

File: src/components/Select/select.module.scss [hunk 1]
```diff
@@ -93,14 +93,14 @@ $multi-select-count-offset: 54px;
     }
 
     .select-clear-button {
-      bottom: 0;
-      height: 44px;
+      bottom: $space-xxs;
+      height: 36px;
```
Comment (jhoward-eightfold) [inline src/components/Select/select.module.scss:97]: why did this get smaller?

File: src/components/Select/Select.types.ts [hunk 2]
```diff
@@ -104,6 +104,11 @@ export interface SelectProps
    * The Select input custom class names.
    */
   inputClassNames?: string;
+  /**
+   * The Select input is readonly.
+   * @default false
+   */
+  inputReadOnly?: boolean;
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.types.ts:111]: does this require the word "input"? would love to dial down the verbosity if possible

File: src/components/Select/Select.tsx [hunk 3]
```diff
@@ -261,6 +262,12 @@ export const Select: FC<SelectProps> = React.forwardRef(
       }
     }, [clear, dropdownVisible, filterable]);
 
+    useEffect(() => {
+      if (inputReadOnly && dropdownVisible) {
+        setDropdownVisibility(false);
```
Comment (jhoward-eightfold) [inline src/components/Select/Select.tsx:267]: nice edge case handling. is it worth a test?

File: src/components/Inputs/TextInput/TextInput.stories.tsx [hunk 4]
```diff
@@ -164,6 +167,12 @@ Text_Input.args = {
   name: 'myTextInput',
   numbersOnly: false,
   placeholder: 'Placeholder text',
+  readonly: {
+    clearable: false,
+    enabled: false,
```
Comment (jhoward-eightfold) [inline src/components/Inputs/TextInput/TextInput.stories.tsx:172]: Is it worthwhile to separate the boolean from the additional props to avoid the type checking?

File: src/components/Inputs/TextArea/TextArea.tsx [hunk 5]
```diff
@@ -217,6 +237,27 @@ export const TextArea: FC<TextAreaProps> = React.forwardRef(
       debouncedChange(e);
     };
 
+    const getIconSize = (): IconSize => {
+      let iconSize: IconSize;
+      if (largeScreenActive) {
```
Comment (jhoward-eightfold) [inline src/components/Inputs/TextArea/TextArea.tsx:242]: would be nice to have a hook which simply returns current screen size ('lg', 'md', 'sm', 'xs' etc), and then a mapping could be used instead of if/else. I would guess this would be useful in other components?

File: src/components/Inputs/Input.types.ts [hunk 6]
```diff
@@ -323,7 +351,7 @@ export interface InputProps<T>
    * input readonly.
    * @default false
    */
-  readonly?: boolean;
+  readonly?: boolean | ReadOnlyProps;
```
Comment (jhoward-eightfold) [inline src/components/Inputs/Input.types.ts:354]: I did this review in reverse order, so this is called out below as well. Can we separate the props and the boolean? It would read nicer without the type checking, etc

File: src/components/Select/select.module.scss (hunk 1 above)
Comment (dkilgore-eightfold) [reply src/components/Select/select.module.scss:97]: Need room for focus-visible rect.

File: src/components/Inputs/TextArea/TextArea.tsx (hunk 5 above)
Comment (dkilgore-eightfold) [reply src/components/Inputs/TextArea/TextArea.tsx:231]: We have such a hook, and this could be a mapping as the properties here are returned by a hook, but will consider this for a future change as it's code style only.

**Review by jhoward-eightfold — APPROVED** [review]
